app/main/views.py

```diff
@@ -49,35 +49,101 @@ def list_services():
         links=pagination_links(services, '.list_services', request.args))
 
 
-@main.route('/services', methods=['POST'])
-def add_service():
-    message = "IDs are currently generated externally, new resources should " \
-              "be created with PUT"
-    return jsonify(error=message), 501
+@main.route('/services-archive', methods=['GET'])
+def list_archived_services_by_service_id():
+    """
+    Retrieves a list of services from the service_archive table
+    for the supplied service_id
+    :query_param service_id:
+    :return: List[service]
+    """
 
+    try:
+        service_id = int(request.args.get("service-id", "no service id"))
+    except ValueError:
+        abort(400, "Invalid service id supplied")
 
-@main.route('/services/<int:service_id>', methods=['PUT'])
+    try:
+        page = int(request.args.get('page', 1))
+    except ValueError:
+        abort(400, "Invalid page argument")
+
+    services = ServiceArchive.query.filter(Service.service_id == service_id)
+
+    services = services.paginate(page=page, per_page=API_FETCH_PAGE_SIZE,
+                                 error_out=False)
+
+    if request.args and not services.items:
+        abort(404)
+    return jsonify(
+        services=list(map(jsonify_service, services.items)),
+        links=pagination_links(services, '.list_services', request.args))
+
+
+@main.route('/services/<int:service_id>', methods=['POST'])
 def update_service(service_id):
+    """
+        Update a service. Looks service up in DB, and updates the JSON listing.
+        Uses existing JSON Parse routines for validation
+    """
+    service = Service.query.filter(
+        Service.service_id == service_id
+    ).first_or_404()
+
+    service_to_archive = prepare_archived_service(service)
+    update_json = get_json_from_request('update_details')['update_details']
+    validate_updater_json_or_400(update_json)
+    service_update = get_json_from_request('services')['services']
+
+    data = dict(service.data.items())
+    data.update(service_update)
+    validate_json_or_400(data)
+
+    now = datetime.now()
+    service.data = data
+    service.updated_at = now
+    service.updated_by = update_json['updated_by']
+    service.updated_reason = update_json['update_reason']
+
+    db.session.add(service)
+    db.session.add(service_to_archive)
+
+    try:
+        db.session.commit()
+    except DatabaseError:
+        db.session.rollback()
+        abort(500, "Database error")
+
+    return jsonify(message="done"), 200
+
+
+@main.route('/services/<int:service_id>', methods=['PUT'])
+def import_service(service_id):
     now = datetime.now()
     service = Service.query.filter(Service.service_id == service_id).first()
-    http_status = 204
-    if service is None:
-        http_status = 201
-        service = Service(service_id=service_id)
-        service.created_at = now
+
+    if service is not None:
+        abort(409, "Service already exists")
```

Comment:
granted. was in my mind deprecating this. on retrospect you're right. will kill this.